app/controllers/api/v3/custom_curves_controller.rb

```diff
@@ -46,16 +46,29 @@ def index
       #
       # GET /api/v3/scenarios/:scenario_id/custom_curves/:name
       def show
-        curve = current_user_curve
+        user_curve = current_user_curve
+        config     = config_for(params[:id])
+        raw_curve  = user_curve.curve
+
+        # Only pull FLH & rescale for CapacityProfile curves
+        if config.processor_key == :capacity_profile
+          flh_key     = config.input_keys.first
+          full_load   = user_curve.scenario.user_values[flh_key]
+          scaled_curve = CurveHandler::Reducers::Rescaler
+                           .new(raw_curve, full_load)
+                           .call
+        else
+          scaled_curve = raw_curve
+        end
```

Comment:
This logic is too complicated for a controller. Remember that we want methods to generally be smaller than 10 lines.